test/sm2_internal_test.c

```diff
@@ -184,7 +184,7 @@ static int test_sm2_crypt(const EC_GROUP *group,
         goto done;
 
     if (!TEST_true(ossl_sm2_plaintext_size(key, digest, ctext_len, &ptext_len))
-            || !TEST_int_eq(ptext_len, msg_len))
+            || !TEST_int_ge(ptext_len, msg_len))
```

Comment:
Why would the plaintext length ever be larger than the message length?

Reply:
int ossl_sm2_plaintext_size(const EC_KEY *key, const EVP_MD *digest,
                            size_t msg_len, size_t *pt_size)
{
    const size_t field_size = ec_field_size(EC_KEY_get0_group(key));
    const int md_size = EVP_MD_get_size(digest);
    size_t overhead;

    if (md_size < 0) {
        ERR_raise(ERR_LIB_SM2, SM2_R_INVALID_DIGEST);
        return 0;
    }
    if (field_size == 0) {
        ERR_raise(ERR_LIB_SM2, SM2_R_INVALID_FIELD);
        return 0;
    }

    **overhead = 10 + 2 * field_size + (size_t)md_size;**
    if (msg_len <= overhead) {
        ERR_raise(ERR_LIB_SM2, SM2_R_INVALID_ENCODING);
        return 0;
    }

    *pt_size = msg_len - overhead;
    return 1;
}

Reply:
int ossl_sm2_plaintext_size(const EC_KEY *key, const EVP_MD *digest,
                            size_t msg_len, size_t *pt_size)
{
    const size_t field_size = ec_field_size(EC_KEY_get0_group(key));
    const int md_size = EVP_MD_get_size(digest);
    size_t overhead;

    if (md_size < 0) {
        ERR_raise(ERR_LIB_SM2, SM2_R_INVALID_DIGEST);
        return 0;
    }
    if (field_size == 0) {
        ERR_raise(ERR_LIB_SM2, SM2_R_INVALID_FIELD);
        return 0;
    }

    overhead = 10 + 2 * field_size + (size_t)md_size;
    if (msg_len <= overhead) {
        ERR_raise(ERR_LIB_SM2, SM2_R_INVALID_ENCODING);
        return 0;
    }

    *pt_size = msg_len - overhead;
    return 1;
}

Reply:
overhead = 10 + 2 * field_size + (size_t)md_size;
10 is not Fixed value, sometimes is larger then 10.
it depends on the asn.1 structure.
in this case, the asn.1 structure len is equal to 11 (not 10).